scripts/code_style_rewrite_branch.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python3
+"""Rewrite a branch on top of a code style change commit so that all
+new commits on the branch look as if they were created after the
+code style change.
+"""
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from pathlib import PurePath
+import subprocess
+import sys
+from typing import List
+
+import code_style
+
+def working_directory_clean() -> bool:
+    """
+    Check if the working directory is clean (i.e. no changes to tracked
+    files staged or unstaged).
+    """
+    # Check for unstaged files
+    result = subprocess.run(["git", "diff", "--exit-code"], \
+            stdout=subprocess.PIPE, check=False)
+    if result.returncode != 0:
+        return False
+    # Check for staged files
+    result = subprocess.run(["git", "diff", "--cached", "--exit-code"], \
+            stdout=subprocess.PIPE, check=False)
+    if result.returncode != 0:
+        return False
+    return True
+
+def delete_branch(branch_name: str) -> bool:
+    """
+    Delete the given branch.
+    """
+    # Go to detached HEAD state in case we are deleting the checked-out branch
+    result = subprocess.run(["git", "checkout", "-d"], check=False)
+    if result.returncode != 0:
+        print("Error detaching HEAD.", file=sys.stderr)
+        return False
+    result = subprocess.run(["git", "branch", "-D", branch_name], check=False)
+    if result.returncode != 0:
+        print("Error deleting branch '" + branch_name + "'.", file=sys.stderr)
+        return False
+    return True
+
+def rebase_on(branch: str, commit: str, new_branch: str) -> bool:
+    """
+    Make a new branch with the given name that consists of the given branch
+    rebased on the given commit.
+    """
+    # Create a new branch from the existing branch
```

Comment:
Minor: this can be done with a single git invocation: `git checkout -b <new_branch> <start_point>`

Reply:
Perhaps not so minor after all: using `git checkout -b <new_branch> <start_point>` would (I think) avoid errors like this when using worktrees:
```
fatal: 'use-psa-rsa-pss' is already checked out at '/home/manpeg01/src/mbedtls/use-psa-rsa-pss'
Error checking out branch 'use-psa-rsa-pss'.
```

Reply:
For now I've made a detached checkout. I might change to `git checkout --no-track -b` (without `--no-track`, we could run into some different surprising behavior).